src/test/java/uk/gov/pay/api/json/CreatePaymentRequestDeserializerTest.java

```diff
@@ -392,20 +431,83 @@ public void deserialize_shouldThrowValidationException_whenAgreementIdIsNotAStri
 
     @Test
     public void deserialize_shouldThrowValidationException_whenAgreementIdIsMoreThan26CharactersLength() throws Exception {
-
-        String json = "{" +
-                "  \"amount\" : 666," +
-                "  \"reference\" : \"Some reference\"," +
-                "  \"description\" : \"Some description\"," +
-                "  \"agreement_id\" : \"" + RandomStringUtils.randomAlphanumeric(27) + "\"" +
+        // language=JSON
+        String json = "{\n" +
+                "  \"amount\": 666,\n" +
+                "  \"reference\": \"Some reference\",\n" +
+                "  \"description\": \"Some description\",\n" +
+                "  \"agreement_id\": \"" + RandomStringUtils.randomAlphanumeric(27) + "\"\n" +
                 "}";
 
         expectedException.expect(aValidationExceptionContaining("P0102", "Invalid attribute value: agreement_id. Must be less than or equal to 26 characters length"));
 
         deserializer.deserialize(jsonFactory.createParser(json), ctx);
     }
-    
-    
+
+    @Test
+    public void deserialize_shouldThrowValidationException_whenLanguageIsNotSupported() throws Exception {
+        // language=JSON
+        String json = "{\n" +
+                "  \"amount\": 1337,\n" +
+                "  \"reference\": \"Some reference\",\n" +
+                "  \"description\": \"Some description\",\n" +
+                "  \"return_url\": \"https://somewhere.gov.uk/rainbow/1\",\n" +
+                "  \"language\": \"fr\"\n" +
+                "}";
+
+        expectedException.expect(aValidationExceptionContaining("P0102", "Invalid attribute value: language. Must be \"en\" or \"cy\""));
```

Comment:
Yeah, I’m the fence about this. The existing tests do use complete strings that hard code maximum amounts and sizes etc., so what we’ve added is compatible with this. Generating the expected error messages from things pulled in from elsewhere in the code makes me a bit wary because there’s the risk that we accidentally change the error messages without realising (which sometimes can be indicative of a buggy behaviour change).